ui/src/app/core/settings.service.ts

```diff
@@ -45,12 +46,14 @@ export class SettingsService {
   /** return the display columns settings for a project or a clean scaffolding, if they don't exist */
   getDisplayColumns(projectId: string): string[] {
     let settings = this.getSettingsForProject(projectId);
-    if (!settings) {
-      settings = this.createEmptySettingsForProject(projectId);
-    }
     return settings.displayColumns;
   }
 
+  getPageSize(projectId: string): number {
+    let settings = this.getSettingsForProject(projectId);
```

Comment:
nit: I wouldn't bother with this local variable.

Reply:
fixed for `getDisplayColumns` and `getPageSize `